includes/extra_configures/enable_error_logging.php

```diff
@@ -12,26 +12,73 @@
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
  * @version $Id: lat9 Sun Feb 25 13:09:11 2018 -0500 Modified in v1.5.6 $
  */
+if (!defined('IS_ADMIN_FLAG')) {
+    exit('Invalid Access');
+}
 
-function zen_debug_error_handler ($errno, $errstr, $errfile, $errline) 
+function zen_debug_error_handler($errno, $errstr, $errfile, $errline) 
 {
     if (!(error_reporting() & $errno)) {
         return;
     }
-    ob_start();
-    if (version_compare(PHP_VERSION, '5.3.6') >= 0) {
-        debug_print_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
-    } else {
-        debug_print_backtrace();
+
+    $handled = true;                    //-For known error types, we'll handle them here.
+    switch ($errno) {
+        case E_NOTICE:
+        case E_USER_NOTICE:
+            $error_type = 'Notice';
+            break;
+        case E_DEPRECATED:
+        case E_USER_DEPRECATED:
+            $error_type = 'Deprecated';
+            break;
+        case E_WARNING:
+        case E_USER_WARNING:
+            $error_type = 'Warning';
+            break;
+        case E_ERROR:
+        case E_USER_ERROR:
+            $error_type = 'Fatal error';
+            break;
+        default:
+            $handled = false;      //-Unknown error type, let PHP's built-in handler do its thing.
+            break;
+    }
+
+    if ($handled) {
```

Comment:
One of the things we're embracing more and more now in ZC code is trying to avoid the arrow-antipattern (unnecessary `if` or nested conditionals), especially where blocks of code are wrapped inside those conditions when an early-return would help make the logic easier to read.

Do you mind updating this to something like:
```php
if (! $handled) {
  return false;
}
// ... the rest of the code that was formerly inside the `if` loop
```

Reply:
No problems; I'll get that updated in a bit.

Reply:
There you go ...